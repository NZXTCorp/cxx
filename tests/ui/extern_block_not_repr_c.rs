mod bindgen {
   pub struct A {
       a: u8,
       b: cxx::CxxString,
   }

   unsafe impl cxx::ExternType for A {
       type Id = cxx::type_id!("A");
       type Kind = cxx::kind::Trivial;
   }
   
}

#[cxx::bridge]
mod ffi {
   extern "C" {
       type A = crate::bindgen::A;

       fn get_a(a: A);
   }
}

fn main() {}
