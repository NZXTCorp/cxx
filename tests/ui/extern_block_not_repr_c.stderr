error: `extern` block uses type `bindgen::A`, which is not FFI-safe
  --> $DIR/extern_block_not_repr_c.rs:19:20
   |
19 |        fn get_a(a: A);
   |                    ^ not FFI-safe
   |
note: the lint level is defined here
  --> $DIR/extern_block_not_repr_c.rs:14:1
   |
14 | #[cxx::bridge]
   | ^^^^^^^^^^^^^^
   = help: consider adding a `#[repr(C)]` or `#[repr(transparent)]` attribute to this struct
   = note: this struct has unspecified layout
note: the type is defined here
  --> $DIR/extern_block_not_repr_c.rs:2:4
   |
2  | /    pub struct A {
3  | |        a: u8,
4  | |        b: cxx::CxxString,
5  | |    }
   | |____^
   = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
